starchiver-ext

```diff
@@ -0,0 +1,484 @@
+            ;;
+        testnet)
+            hashurl="http://5.161.189.245:7777/hash.txt"
+            ;;
+        *)
+            echo "Invalid network choice: $network_choice"
+            exit 1
+            ;;
+    esac
+}
+
+# Main menu function
+main_menu() {
+    while true; do
+        clear
+        echo ""
+        echo ""
+        talk "Don't forget to tip the bar tender!" $BOLD$YELLOW
+        talk "  ${BOLD}This script was written by:${NC} ${BOLD}${LGREEN}@Proph151Music${NC}"
+        talk "     ${BOLD}for the ${LBLUE}Constellation Network${NC} ${BOLD}ecosystem.${NC}"
+        echo ""
+        talk "  DAG Wallet Address for sending tips can be found here..." $YELLOW
+        talk "     ${BOLD}DAG0Zyq8XPnDKRB3wZaFcFHjL4seCLSDtHbUcYq3${NC}"
+        echo ""
+        talk "---==[ STARCHIVER ]==---" $BOLD$LGREEN
+        talk "Create and Restore Starchive files." $LGREEN
+        echo ""
+        talk "Select a network:"
+        talk "${BOLD}M)${NC} ${BOLD}${LCYAN}MainNet${NC}"
+        talk "${BOLD}I)${NC} ${BOLD}${LCYAN}IntegrationNet${NC}"
+        talk "${BOLD}T)${NC} ${BOLD}${LCYAN}TestNet${NC}"
+        talk "${BOLD}C)${NC} ${BOLD}${LCYAN}Custom${NC}"
+        talk "${BOLD}Q)${NC} ${BOLD}${LCYAN}Quit${NC}"
+        echo ""
+        read -p "$(echo -e ${BOLD}Choose your adventure${NC} [M, I, T, C, Q]:) " network_choice
+        echo ""
+
+        case $network_choice in
+            [Mm])
+                path=$(search_data_folders | xargs)
+                if [ $? -eq 1 ] || [ -z "$path" ]; then
+                    talk "No valid data folder with snapshot selected. Exiting." $LRED
+                    exit 1
+                fi
+                hashurl="http://5.161.53.65:7777/hash.txt"
+                download_verify_extract_tar "$hashurl" "$path"
+                ;;
+            [Ii])
+                path=$(search_data_folders | xargs)
+                if [ $? -eq 1 ] || [ -z "$path" ]; then
+                    talk "No valid data folder with snapshot selected. Exiting." $LRED
+                    exit 1
+                fi
+                hashurl="http://5.161.93.223:7777/hash.txt"
+                download_verify_extract_tar "$hashurl" "$path"
+                ;;
+            [Tt])
+                path=$(search_data_folders | xargs)
+                if [ $? -eq 1 ] || [ -z "$path" ]; then
+                    talk "No valid data folder with snapshot selected. Exiting." $LRED
+                    exit 1
+                fi
+                hashurl="http://5.161.189.245:7777/hash.txt"
+                download_verify_extract_tar "$hashurl" "$path"
+                ;;
+            [Cc])
+                path=$(search_data_folders | xargs)
+                if [ $? -eq 1 ] || [ -z "$path" ]; then
+                    talk "No valid data folder with snapshot selected. Exiting." $LRED
+                    exit 1
+                fi
+                read -p "${BOLD}Enter the URL of the Hash file:${NC}) " hashurl
```

Comment:
It would be great if this could be passed via a cli arg also. That would mean this could be automated without waiting for an interactive response.